servers/cromwell/jobs/__main__.py

```diff
@@ -31,13 +33,42 @@
     args, _ = parser.parse_known_args()
 
 app = connexion.App(__name__, specification_dir='./swagger/', swagger_ui=False)
+DEFAULT_CROMWELL_CREDENTIALS = {"cromwell_user": "", "cromwell_password": ""}
+LOADER_ERR_MSG = "Failed to load config.json, using the default config: {0}".format(
+    DEFAULT_CROMWELL_CREDENTIALS)
+
+# Load credentials for cromwell
 config_path = os.environ.get('CROMWELL_CREDENTIALS')
-with open(config_path) as f:
-    config = json.load(f)
+
+# Check if the credentials are provided properly
+try:
+    with open(config_path) as f:
+        config = json.load(f)
+except (IOError, TypeError):
+    print(LOADER_ERR_MSG)
+    config = DEFAULT_CROMWELL_CREDENTIALS
+finally:
     app.app.config.update(config)
+
 app.app.config['cromwell_url'] = args.cromwell_url
 app.app.json_encoder = JSONEncoder
 app.add_api('swagger.yaml', base_path=args.path_prefix)
 
-if __name__ == '__main__':
-    app.run(host='0.0.0.0', port=args.port)
+
+def run():
+    # Check the connections with cromwell
+    try:
+        response = requests.head(
+            args.cromwell_url,
+            auth=HTTPBasicAuth(app.app.config["cromwell_user"],
+                               app.app.config["cromwell_password"]),
+            timeout=5)
+        if response.status_code == 401:
+            raise requests.exceptions.HTTPError(
+                "Invalid credentials for the Cromwell: {0}".format(
+                    args.cromwell_url))
+        else:
+            return app.app
+    except requests.exceptions.RequestException as err:
+        print(err)
```

Comment:
How about using logging instead of printing these errors?

Reply:
That's a good idea! Will refactor it, thanks!